src/lib/chat/matrix-client.ts

```diff
@@ -496,14 +499,31 @@ export class MatrixClient implements IChatClient {
     }
   }
 
-  private initializeRoomEventHandlers(room: Room) {
+  private async processRoomTimelineEvent(
+    event: MatrixEvent,
+    room: Room | undefined,
+    toStartOfTimeline: boolean | undefined,
+    removed: boolean,
+    data: IRoomTimelineData
+  ) {
+    if (removed) return; // only notify for new events, not removed ones
+    if (!data.liveEvent || !!toStartOfTimeline) return; // only notify for new things, not old.
+    if (!this.isSyncing) return; // only notify if syncing is complete
+    if (event.getSender() === this.userId) return; // only notify for other users events
```

Comment:
Do you mean ` if (event.getSender() === this.userId)` ? this would prevent the user receiving browser notifications when sending a message themselves.

Reply:
Yes, but the matrix client shouldn't decide what our app cares or does not care about here. It should publish this event no matter who it's related to. Our browser notification saga can then filter out events it doesn't care about.

Reply:
Nice yeah makes sense to remove the check here. I adjusted the `isOwner` check in the `sendBrowserNotification` saga function so it actually works and handles this check correctly.